src/main/keyboard/index.js

```diff
@@ -1,56 +1,76 @@
-// import EventEmitter from 'events'
-import { getCurrentKeyboardLayout, getCurrentKeyboardLanguage, getCurrentKeymap } from 'keyboard-layout'
+import { ipcMain, shell } from 'electron'
+import log from 'electron-log'
+import EventEmitter from 'events'
+import fsPromises from 'fs/promises'
+import { getCurrentKeyboardLayout, getKeyMap, onDidChangeKeyboardLayout } from 'native-keymap'
+import os from 'os'
+import path from 'path'
 
-export const getKeyboardLanguage = () => {
-  const lang = getCurrentKeyboardLanguage()
-  if (lang.length >= 2) {
-    return lang.substring(0, 2)
+let currentKeyboardInfo = null
+const loadKeyboardInfo = () => {
+  currentKeyboardInfo = {
+    layout: getCurrentKeyboardLayout(),
+    keymap: getKeyMap()
   }
-  return lang
+  return currentKeyboardInfo
 }
 
-export const dumpKeyboardInformation = () => {
-  return `Layout: ${getCurrentKeyboardLayout()}\n` +
-    `Language: ${getCurrentKeyboardLanguage()}\n\n` +
-    JSON.stringify(getCurrentKeymap(), null, 2)
+export const getKeyboardInfo = () => {
+  if (!currentKeyboardInfo) {
```

Comment:
I think put this `if statements` in `loadKeyboardInfo` is much better, and use like this:

```js
return loadKeyboardInfo()
```

Reply:
`loadKeyboardInfo` is used to load the current key layout from the system but `getKeyboardInfo` should only return the cache value.